src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -47,44 +58,48 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
                     : nameof(originalTemplateRoot));
             }
 
-            // Attempt to find an equivalent JToken in the original template from the expanded template's path directly
-            var tokenFromOriginalTemplate = originalTemplateRoot.InsensitiveToken(pathInExpandedTemplate, InsensitivePathNotFoundBehavior.LastValid);
+            JToken tokenFromOriginalTemplate;
 
-            // If the JToken returned from looking up the expanded template path is
-            // just pointing to the root of the original template, then
-            // even the first property could not be found in the original template.
-            if (tokenFromOriginalTemplate.Equals(originalTemplateRoot))
-            {
-                return 1;
-            }
+            // Handle path and prefixes backwards one level at a time to construct an accurate resources' path
+            var currentContext = this.templateContext;
+            var currentPathToEvaluate = pathInExpandedTemplate;
+            string mappedPathInExpandedTemplate = "";
 
-            // If the path is in the resources array of the template
-            var matches = resourceIndexInPath.Matches(pathInExpandedTemplate);
-            if (matches.Count > 0)
+            while (currentContext != null && currentPathToEvaluate.Length > 0)
             {
-                // Get the path of the child resource in the expanded template
-                string resourceWithIndex = string.Join('.', matches);
-
-                // Verify the expanded template is available.
-                // (Avoid throwing earlier since this is not always needed.)
-                if (expandedTemplateRoot == null)
+                // If the path is in the resources array of the template
+                var matches = resourceIndexInPath.Matches(currentPathToEvaluate);
+                if (matches.Count > 0)
                 {
-                    throw new ArgumentNullException(nameof(expandedTemplateRoot));
-                }
+                    // Get the path of the child resource in the expanded template
+                    string resourceWithIndex = string.Join('.', matches);
 
-                string remainingPathAtResourceScope = pathInExpandedTemplate[(resourceWithIndex.Length + 1)..];
+                    // Verify the expanded template is available.
+                    // (Avoid throwing earlier since this is not always needed.)
+                    if (expandedTemplateRoot == null)
+                    {
+                        throw new ArgumentNullException(nameof(expandedTemplateRoot));
```

Comment:
Can you also add more information on this exception message?

Reply:
This wasn't my addition so I don't really know why I think Nick, made the decision to throw this at this point. I'm not sure if that is what you are referring to